### PR TITLE
EPMRPP-33459 React. Dashboard. Widget. Unique bug table

### PR DESCRIPTION
**Pull request by andrei-bulagin**

### CODE REVIEW

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import TagIcon from 'common/img/tag-inline.svg';
+import Parser from 'html-react-parser';
+import styles from './foundIn.scss';
+
+export const cx = classNames.bind(styles);
+
+@connect((state) => ({
+  projectId: activeProjectSelector(state),
+}))
+export class FoundIn extends PureComponent {
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx]: Connected components are pure by default.

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return (
+      <div className={cx('found-in-col', className)}>
+        {value.foundIn &&
+          value.foundIn.map((item) => {
+            const path = [item.launchId].concat(Object.keys(item.pathNames)).join('/');
+            return (
+              <Fragment key={item.id}>
+                <div className={cx('item')} title={item.name}>
+                  <Link
+                    to={`#${projectId}/launches/all/${path}?log.item=${item.id}`}
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx]: We encapsulare all paths in the router config, so please use objects for links.
```
{
  type: PROJECT_LOG_PAGE,
  payload: {
    projectId,
    filterId: ALL, // from app/src/common/constants/reservedFilterIds.js
    testItemIds: path, // const path = [item.launchId].concat(Object.keys(item.pathNames), item.id).join('/');
  }
}
```

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.found-in-col {
+  width: 40%;
+  font-size: 13px;
+  line-height: 1.42857143;
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.scss]: This number looks strange

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.stories.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { storiesOf } from '@storybook/react';
+import { host } from 'storybook-host';
+import { withReadme } from 'storybook-readme';
+import 'c3/c3.css';
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.stories.jsx]: It's not necessary. This import has been moved to the storybook config.

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/messages.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+);
+
+const submitterColumnTitle = (
+  <FormattedMessage id={'UniqueBugsTable.submitterColumn'} defaultMessage={'SUBMITTER'} />
+);
+
+export const COLUMN_NAMES_MAP = {
+  [BUG_ID]: {
+    full: bugIDColumnTitle,
+    short: bugIDColumnTitle,
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/messages.jsx]: No need to duplicate titles. If short one is not present, full title is used.

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.jsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+const columnComponentsMap = {
+  [BUG_ID_COLUMN_KEY]: BugIDColumn,
+  [FOUND_IN_COLUMN_KEY]: FoundInColumn,
+  [SUBMIT_DATE_COLUMN_KEY]: SubmitDateColumn,
+  [SUBMITTER_COLUMN_KEY]: SubmitterColumn,
+};
+
+const getColumn = (name, columnType, formatMessage) => ({
+  id: name,
+  title: COLUMN_NAMES_MAP[name],
```
Comment (DiscoElevator) [inline app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.jsx]: Is it possible to do this?
`title: formatMessage(COLUMN_NAMES_MAP[name]),`

File: app/src/components/main/grid/propTypes.js [hunk 7]
```diff
@@ -3,8 +3,8 @@ import { ALIGN_CENTER, ALIGN_LEFT, ALIGN_RIGHT } from './constants';
 
 export const columnPropTypes = {
   title: PropTypes.shape({
-    full: PropTypes.string,
-    short: PropTypes.string,
+    full: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
```
Comment (DiscoElevator) [inline app/src/components/main/grid/propTypes.js]: Looks like this change is not necessary for now. See comments below.

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx (hunk 1 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx]: Changed

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx (hunk 2 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.jsx]: Fixed

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.scss (hunk 3 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/foundIn/foundIn.scss]: Replaced to fixed number

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.stories.jsx (hunk 4 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.stories.jsx]: Removed

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/messages.jsx (hunk 5 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/messages.jsx]: Removed

File: app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.jsx (hunk 6 above)
Comment (andrei-bulagin) [reply app/src/pages/inside/dashboardPage/widgets/uniqueBugsTable/uniqueBugsTable.jsx]: Yes, replaced to `title: { full: formatMessage(COLUMN_NAMES_MAP[name]) },`

File: app/src/components/main/grid/propTypes.js (hunk 7 above)
Comment (andrei-bulagin) [reply app/src/components/main/grid/propTypes.js]: Reverted

**Pull request merged**